packages/twenty-server/src/modules/workflow/common/workspace-services/workflow-version.workspace-service.ts

```
@@ -0,0 +1,289 @@
+        'workflowVersion',
+      );
+
+    const workflowVersion = await workflowVersionRepository.findOneOrFail({
+      where: {
+        id: workflowVersionId,
+      },
+    });
+
+    await workflowVersionRepository.update(workflowVersion.id, {
+      steps: [...(workflowVersion.steps || []), enrichedNewStep],
+    });
+
+    return enrichedNewStep;
+  }
+
+  async updateWorkflowVersionStep({
+    workspaceId,
+    workflowVersionId,
+    step,
+  }: {
+    workspaceId: string;
+    workflowVersionId: string;
+    step: WorkflowAction;
+  }): Promise<WorkflowAction> {
+    const workflowVersionRepository =
+      await this.twentyORMManager.getRepository<WorkflowVersionWorkspaceEntity>(
+        'workflowVersion',
+      );
+
+    const workflowVersion = await workflowVersionRepository.findOneOrFail({
+      where: {
+        id: workflowVersionId,
+      },
+    });
+
+    if (!isDefined(workflowVersion.steps)) {
+      throw new Error("Can't update step from undefined steps");
+    }
+
+    const enrichedNewStep = await this.enrichOutputSchema({
+      step,
+      workspaceId,
+    });
+
+    const updatedSteps = workflowVersion.steps.map((existingStep) => {
+      if (existingStep.id === step.id) {
+        return enrichedNewStep;
+      } else {
+        return existingStep;
+      }
+    });
+
+    await workflowVersionRepository.update(workflowVersion.id, {
+      steps: updatedSteps,
+    });
+
+    return enrichedNewStep;
+  }
+
+  async deleteWorkflowVersionStep({
+    workspaceId,
+    workflowVersionId,
+    stepId,
+  }: {
+    workspaceId: string;
+    workflowVersionId: string;
+    stepId: string;
+  }): Promise<WorkflowActionDTO> {
+    const workflowVersionRepository =
+      await this.twentyORMManager.getRepository<WorkflowVersionWorkspaceEntity>(
+        'workflowVersion',
+      );
+
+    const workflowVersion = await workflowVersionRepository.findOneOrFail({
```

can we throw a specific exception 🙏